rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1746,28 +1753,47 @@ public Response createStudentContactLogEntry(@PathParam("ID") Long id, fi.otavan
       return Response.status(Status.FORBIDDEN).build();
     }
     
+    User loggedUser = sessionController.getUser();
+    StaffMember creator = userController.findStaffMemberById(loggedUser.getId());
+    
     StudentContactLogEntryType type = entity.getType() != null ? StudentContactLogEntryType.valueOf(entity.getType().name()) : null;
     StudentContactLogEntry contactLogEntry = studentContactLogEntryController.createContactLogEntry(student, type, entity.getText(),
-        toDate(entity.getEntryDate()), entity.getCreatorName());
+        toDate(entity.getEntryDate()), loggedUser.getFirstName() + " " + loggedUser.getLastName(), creator);
 
     return Response.ok(objectFactory.createModel(contactLogEntry)).build();
   }
 
   @Path("/students/{STUDENTID:[0-9]*}/contactLogEntries")
   @GET
-  @RESTPermit(StudentContactLogEntryPermissions.LIST_STUDENTCONTACTLOGENTRIES)
+  @RESTPermit(handling = Handling.INLINE)
   public Response listStudentContactLogEntriesByStudent(@PathParam("STUDENTID") Long studentId) {
     Student student = studentController.findStudentById(studentId);
     Status studentStatus = checkStudent(student);
     if (studentStatus != Status.OK)
-      return Response.status(studentStatus).build();
-
-    return Response.ok(objectFactory.createModel(studentContactLogEntryController.listContactLogEntriesByStudent(student))).build();
+      return Response.status(studentStatus).build(); 
+    
+    User loggedUser = sessionController.getUser();
+    
+    StaffMember staffMember = userController.findStaffMemberById(loggedUser.getId());
+    
+    if (staffMember == null) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    ContactLogAccess access = studentController.resolveContactLogAccess(student);
+    
+    if (access.equals(ContactLogAccess.ALL)) {
+      return Response.ok(objectFactory.createModel(studentContactLogEntryController.listContactLogEntriesByStudent(student))).build();
+    } else if (access.equals(ContactLogAccess.OWN)) {
+      return Response.ok(objectFactory.createModel(studentContactLogEntryController.listContactLogEntriesByStudentAndCreator(student, staffMember))).build();
+    } else {
+      return Response.status(Status.FORBIDDEN).build();
+    }
   }
 
   @Path("/students/{STUDENTID:[0-9]*}/contactLogEntries/{ID:[0-9]*}")
   @GET
-  @RESTPermit(StudentContactLogEntryPermissions.FIND_STUDENTCONTACTLOGENTRY)
+  @RESTPermit(handling = Handling.INLINE)
   public Response findStudentContactLogEntryById(@PathParam("STUDENTID") Long studentId, @PathParam("ID") Long id) {
```

Comment:
Shouldn't this endpoint have the `ContactLogAccess access = studentController.resolveContactLogAccess(student);` check too